crates/compiler/load/tests/test_reporting.rs

```diff
@@ -14908,6 +15016,141 @@ All branches in an `if` must have the same type!
         @"" // no errors
     );
 
+    test_report!(
+        try_with_non_result_target,
+        indoc!(
+            r#"
+            invalidTry = \{} ->
+                x = try 64
+
+                Ok (x * 2)
+
+            invalidTry {}
+            "#
+        ),
+        @r"
+    ── INVALID TRY TARGET in /code/proj/Main.roc ───────────────────────────────────
+
+    This expression cannot be used as a `try` target:
+
+    5│          x = try 64
+                        ^^
+
+    I expected a Result, but it actually has type:
+
+        Num *
+
+    Hint: Did you forget to wrap the value with an `Ok` or an `Err` tag?
+    "
+    );
+
+    test_report!(
+        incompatible_try_errs,
+        indoc!(
+            r#"
+            incompatibleTrys = \{} ->
+                x = try Err 123
+
+                y = try Err "abc"
+
+                Ok (x + y)
+
+            incompatibleTrys {}
+            "#
+        ),
+        @r#"
+    ── TYPE MISMATCH in /code/proj/Main.roc ────────────────────────────────────────
+
+    This returns something that's incompatible with the return type of the
+    enclosing function:
+
+    5│           x = try Err 123
+    6│
+    7│>          y = try Err "abc"
+    8│
+    9│           Ok (x + y)
+
+    This returns an `Err` of type:
+
+        [Err Str, …]
+
+    But I expected the function to have return type:
+
+        [Err (Num *), …]a
+    "#
+    );
+
+    test_report!(
+        try_prefix_in_pipe,
+        indoc!(
+            r#"
+            readFile : Str -> Str
+
+            getFileContents : Str -> Result Str _
+            getFileContents = \filePath ->
+                contents =
+                    readFile filePath
+                    |> try Result.mapErr ErrWrapper
+
+                contents
+
+            getFileContents "file.txt"
+            "#
+        ),
+        @r"
+    ── TYPE MISMATCH in /code/proj/Main.roc ────────────────────────────────────────
+
+    This 1st argument to this function has an unexpected type:
+
+     9│>              readFile filePath
```

Comment:
This validates that we see the full pipe instead of just the `readFile filePath` arg, which was confusing.